src/views/SignTransactionLedger.vue

```diff
@@ -175,17 +202,32 @@ export default class SignTransactionLedger extends Vue {
     private async mounted() {
         const network = this.$refs.network as Network;
 
+        let recipient;
+        let value;
+        let fee;
         let validityStartHeight;
+        let data;
+        let flags;
         if (this.request.kind === RequestType.SIGN_TRANSACTION) {
             const signTransactionRequest = this.request as ParsedSignTransactionRequest;
-            validityStartHeight = signTransactionRequest.validityStartHeight;
+            ({ recipient, value, fee, validityStartHeight, data, flags } = signTransactionRequest);
         } else if (this.request.kind === RequestType.CHECKOUT) {
             const checkoutRequest = this.request as ParsedCheckoutRequest;
+            ({ recipient, value, fee, data, flags } = checkoutRequest);
             const blockchainHeight = await network.getBlockchainHeight(); // usually instant as synced in checkout
             // The next block is the earliest for which tx are accepted by standard miners
             validityStartHeight = blockchainHeight + 1
                 - TX_VALIDITY_WINDOW
                 + checkoutRequest.validityDuration;
+        } else if (this.request.kind === RequestType.CASHLINK) {
+            if (!this.cashlink) {
+                // already handled in created
+                return;
+            }
```

Comment:
If it's already handled, why is this necessary here? Only for TS to not complain?

Reply:
`mounted` can execute before the `reject` closes the window. Therefore I need to stop the execution here.

Reply:
Could you put that in the comment at the next opportunity? :slightly_smiling_face:

Reply:
The changes for checkout/ledger move all the logic to the `mounted` hook which makes this clearer.